crates/node/src/networking/p2p.rs

```diff
@@ -111,6 +124,122 @@ impl Handshake for P2P {
+                stream.write_all(&peer_list_bytes).await?;
+
+                // receive the peer list
+                let buffer_len = stream.read_u32().await? as usize;
+                //TODO validate buffer lengh
+                let mut buffer = vec![0; buffer_len];
+                stream.read_exact(&mut buffer).await?;
+                let distant_peer_list: BTreeSet<SocketAddr> = bincode::deserialize(&buffer)
+                    .map_err(|err| {
+                        std::io::Error::new(
+                            std::io::ErrorKind::Other,
+                            format!("deserialize error:{err}"),
+                        )
+                    })?;
+
+                self.peer_addr_mapping
+                    .write()
+                    .insert(stream.peer_addr().unwrap(), stream.peer_addr().unwrap());
+
+                (distant_peer_list, stream.peer_addr().unwrap())
+            }
+            ConnectionSide::Responder => {
+                //receive the connecting node addr
+                let buffer_len = stream.read_u32().await? as usize;
+                let mut buffer = vec![0; buffer_len];
+                stream.read_exact(&mut buffer).await?;
+                let distant_listening_addr: SocketAddr =
+                    bincode::deserialize(&buffer).map_err(|err| {
+                        std::io::Error::new(
+                            std::io::ErrorKind::Other,
+                            format!("deserialize error:{err}"),
+                        )
+                    })?;
+                {
+                    self.peer_list.write().insert(distant_listening_addr);
+                    self.peer_addr_mapping
+                        .write()
+                        .insert(stream.peer_addr().unwrap(), distant_listening_addr);
+                }
+
+                // receive the peer list
+                let buffer_len = stream.read_u32().await? as usize;
+                let mut buffer = vec![0; buffer_len];
+                stream.read_exact(&mut buffer).await?;
+                let distant_peer_list: BTreeSet<SocketAddr> = bincode::deserialize(&buffer)
+                    .map_err(|err| {
+                        std::io::Error::new(
+                            std::io::ErrorKind::Other,
+                            format!("deserialize error:{err}"),
+                        )
+                    })?;
+
+                //send peer list
+                stream.write_u32(peer_list_bytes.len() as u32).await?;
+                stream.write_all(&peer_list_bytes).await?;
+
+                (distant_peer_list, distant_listening_addr)
+            }
+        };
+
+        //do peer comparition
+        let mut local_diff = {
+            let local_peer_list: &mut BTreeSet<SocketAddr> = &mut self.peer_list.write();
+            let distant_diff: BTreeSet<SocketAddr> = local_peer_list
+                .difference(&distant_peer_list)
+                .cloned()
+                .collect();
```

Comment:
```suggestion
```
For all I can read & test, this seems to be unused?

I wonder why the compiler / linters don't highlight this?

Reply:
Yes it's not use and strange it's not detected. I'll remove it